### PR TITLE
Move from rest-hooks to react-query

### LINKED ISSUE #10400: Replace rest-hooks by react-query
We currently use a mix of `rest-hooks` and `react-query` in the frontend. We want to align this to only one query library and decided to move to react-query going forward. We should replace all existing usage of `rest-hooks` with `react-query`.

### PR DESCRIPTION
**Pull request by jamakase**
## What
Last piece that finally fixes #10400

- removes `rest-hook` library :tada:
- removes unnecessary service injections
- removes `rest-hook` from tests and storybook
- sets set initialSetupComplete in `/workspaces/get` call instead of `workspaces/list` call in e2e tests.

### CODE REVIEW

File: airbyte-webapp/src/hooks/services/useConnector.test.tsx [hunk 1]
```diff
@@ -1,93 +1,71 @@
...
-        },
-      ],
-    },
-  },
-];
+jest.mock("services/connector/DestinationDefinitionService", () => ({
+  useDestinationDefinitionList: () => ({
+    destinationDefinitions: [
+      {
+        destinationDefinitionId: "sid1",
```
Comment (timroes) [inline airbyte-webapp/src/hooks/services/useConnector.test.tsx:28]: Trying to understand this change (and the same a few lines below): The tests in this file are all skipped for now anyway, is there any specific reason we changed those IDs to overlap with the sourceDefinitionIds above? Or is this some artifact from trying to get those tests unskipped again?

**Review by timroes — COMMENTED** [review]
(leaving some half-way-through comments, while reviewing the remaining files)

File: airbyte-webapp/src/views/common/StoreProvider.tsx [hunk 2]
```diff
@@ -1,21 +1,19 @@
...
 import React from "react";
-import { CacheProvider } from "rest-hooks";
 
 const queryClient = new QueryClient({
   defaultOptions: {
     queries: {
       suspense: true,
       refetchOnWindowFocus: false,
       refetchOnReconnect: false,
+      retry: 0,
```
[the remark this answers is not in the document]
Comment (timroes) [reply airbyte-webapp/src/views/common/StoreProvider.tsx:10]: Sounds good to me!

File: airbyte-webapp/src/hooks/services/useConnector.test.tsx (hunk 1 above)
Comment (Jamakase) [reply airbyte-webapp/src/hooks/services/useConnector.test.tsx:28]: This test was disabled before. I tried to make it as close to working as possible, but the issue is that it is difficult to test that `useUpdateDestinationDefinition` was called, as we need to mock mutateAsync. I thought, that we need to improve DI a bit, so we could inject `DefinitionService` with mocks later and check the call there.

Right now, it is not that easy to fix this test ( or a bit more investigation is required ).

File: airbyte-webapp-e2e-tests/cypress/support/commands/workspaces.js [hunk 3]
```diff
@@ -1,9 +1,9 @@
 Cypress.Commands.add("initialSetupCompleted", (completed = true) => {
   // Modify the workspaces/list response to mark every workspace as "initialSetupComplete" to ensure we're not showing
   // the setup/preference page for any workspace if this method got called.
-  cy.intercept("POST", "/api/v1/workspaces/list", (req) => {
+  cy.intercept("POST", "/api/v1/workspaces/get", (req) => {
     req.continue(res => {
-      res.body.workspaces = res.body.workspaces.map(ws => ({ ...ws, initialSetupComplete: completed }));
+      res.body.workspaces = res.body.initialSetupComplete = completed;
```
[the remark this answers is not in the document]
Comment (timroes) [reply airbyte-webapp-e2e-tests/cypress/support/commands/workspaces.js:6]: Sorry, I am not sure I understand that answer fully. I think I get why this line was there beforehand. But right now this mock just intercepts the workspace/get call, those response should not have a `workspaces` parameter in it'sbody at all, that we're setting here?

[the remark this answers is not in the document]
Comment (Jamakase) [reply airbyte-webapp-e2e-tests/cypress/support/commands/workspaces.js:6]: Oh, it was assigned to `res.body.workspaces` too. I haven't noticed it.

**Comment by timroes** [comment]
There is currently one a bit weird behavior, where loading the connection list in one workspace and switching over to another workspace and going back to the connection page (of course works with other resources the same way), will shortly show the connection list of the old workspace before updating them. I think we'll still need to make sure that all queries are invaldiated once the workspace is switched.

**Comment by Jamakase** [comment]
> There is currently one a bit weird behavior, where loading the connection list in one workspace and switching over to another workspace and going back to the connection page (of course works with other resources the same way), will shortly show the connection list of the old workspace before updating them. I think we'll still need to make sure that all queries are invaldiated once the workspace is switched.

Hmm, indeed, we may need to invalidate all `list` queries. Do you think we should prepend all workspace specific queries with `workspaceId` in filters, or we should just invalidate all queries?

**Comment by timroes** [comment]
I think it would make sense adding the workspace ID to the query key for at least all list queries. That said I'd still additionally invalidate all queries I think when switching, since we have most things scoped to workspaces for now (definitions aren't yet,but that will change soon), so I don't think it's a huge problem invalidating all queries, and leaves us on the safe side of not accidentally missing a query.

**Comment by Jamakase** [comment]
> I think it would make sense adding the workspace ID to the query key for at least all list queries. That said I'd still additionally invalidate all queries I think when switching, since we have most things scoped to workspaces for now (definitions aren't yet,but that will change soon), so I don't think it's a huge problem invalidating all queries, and leaves us on the safe side of not accidentally missing a query.

As per our discussion offline - added invalidation of scopes. If required - later workspaceId could be added ( in case smoother behaviour is preferred while switching between workspaces ). 

Also there was an issue, that queries were reset instead of deleting on changing workspaces.

**Review by timroes — APPROVED** [review]
Tested every page locally, everything seems to work as expected. Looked through the code, looks good to me.

**Pull request merged**